Fixes Comgr build warning seen in CI Release builds (gcc):

```
[98/107] Building CXX object CMakeFiles/amd_comgr.dir/src/comgr-hotswap-patch-trampoline.cpp.o
In file included from llvm-project/llvm/include/llvm/MC/MCSection.h:21,
                 from llvm-project/llvm/include/llvm/MC/MCContext.h:24,
                 from llvm-project/amd/comgr/src/comgr-hotswap-internal.h:42,
                 from llvm-project/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp:16:
In member function 'bool llvm::MCOperand::isImm() const',
    inlined from 'bool COMGR::hotswap::bumpNextWaitDscnt(PatchContext&, size_t)' at llvm-project/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp:299:18,
    inlined from 'bool COMGR::hotswap::patchDs2AddrStride64(PatchContext&, size_t)' at llvm-project/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp:350:20:
llvm-project/llvm/include/llvm/MC/MCInst.h:66:31: warning: '*(const llvm::MCOperand*)((char*)&NewInst + offsetof(llvm::MCInst, llvm::MCInst::Operands.llvm::SmallVector<llvm::MCOperand, 6>::<unnamed>.llvm::SmallVectorStorage<llvm::MCOperand, 6>::InlineElts[0])).llvm::MCOperand::Kind' may be used uninitialized [-Wmaybe-uninitialized]
   66 |   bool isImm() const { return Kind == kImmediate; }
      |                               ^~~~
llvm-project/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp: In function 'bool COMGR::hotswap::patchDs2AddrStride64(PatchContext&, size_t)':
llvm-project/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp:297:12: note: 'NewInst' declared here
  297 |     MCInst NewInst = DI.Inst;
```

--- a/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp
+++ b/amd/comgr/src/comgr-hotswap-patch-trampoline.cpp
@@ -294,9 +294,9 @@ static bool bumpNextWaitDscnt(PatchContext &Ctx, size_t Idx) {
 
     // s_wait_dscnt has a single immediate operand (the wait count) at
     // index 0. Increment it directly.
-    MCInst NewInst = DI.Inst;
-    if (NewInst.getNumOperands() == 0)
+    if (DI.Inst.getNumOperands() == 0)
       return false;
+    MCInst NewInst = DI.Inst;
     MCOperand &Op = NewInst.getOperand(0);
     if (!Op.isImm())
       return false;

